src/state/MediaViewModel.ts

```diff
@@ -243,69 +250,72 @@ abstract class BaseMediaViewModel extends ViewModel {
       [audio, this.video],
       (a, v) =>
         encryptionSystem.kind !== E2eeType.NONE &&
-        (a.publication?.isEncrypted === false ||
-          v.publication?.isEncrypted === false),
-    ).pipe(
-      distinctUntilChanged(),
-      shareReplay({ bufferSize: 1, refCount: false }),
+        (a?.publication?.isEncrypted === false ||
+          v?.publication?.isEncrypted === false),
+    ).pipe(this.scope.state());
+
+    this.encryptionStatus = this.participant.pipe(
+      switchMap((participant): Observable<EncryptionStatus> => {
+        if (!participant) {
+          return of(EncryptionStatus.Connecting);
```

Comment:
If we include this state, I am wondering if `EncryptionStatus` is the right type name.
I think this could be considered the overall state of a rtc member once they have an rtc session.
```
  Connecting,
  Okay,
  KeyMissing,
  KeyInvalid,
  PasswordInvalid,
```
On the base rtc membership we have: the lk participant, the media key used by the participant and the keys we have received over matrix.
Thos are the possible states:
lk | matrix key | matches lk key | implies state
--- | --- | --- | ---
false | true/false | true/false | `Connecting`
true | false | false/false | `KeyMissing`
true | true | false | `KeyInvalid`
true | true | true | `Okay`

Looking at it like this I am wondering if the logic becomes much simpler if we first fo 3 bool observables (lk, matrix key, matches lk key) (or 1 observable with an object that has those 3 props) and then build one state observable based on those precomputed observables. The individual bool observables can also be helpful I think:
```
let gotKey = encryptionState === KeyInvalid || encryptionState === Okay
```
Would be used otherwise for checking for a received matrix key.

Reply:
Given that this value isn't currently being used, I think we can move this clean up to a new PR at a later date.

Reply:
What about renaming `encryptionStatus` to `rtcMediaState`